### DISCUSSION TITLE
Error while redirecting to a new page

### DISCUSSION THREAD

**Discussion by junaidfactscomputer** [vercel/next.js#50645] [Help]
### Summary

when redirecting during testing it was working but after hosted giving error Bad Request - Invalid URL
HTTP Error 400. The request URL is invalid. for the below url
**localhost:3000/paymentredirect/@77539A0978176A5160ACDF87F51A9D6C2AC1A6885732571B87E2D20AE7718F19D0FDD32388B67377941D91E791934C6D1F064DE20F445E05BCC4B7BA6539C93A355C7641C3740192C1DD6038841BFB0C26B7FB01D1A4543616768D2191D6F7B86B853B6E90882EC27A1FCF1E9834E02955443BED9318947FDC4F3EF4CFCD546BE7DB3D782C87923DDBD2C0035AD52C6368D0A0E9B5CDF5673406E642264250A3ACA5F192EAE4B85494E6521DC0E873B8E9B2508A4AAA73A6C004EF7BB45A2DDB4ECA5BEC72F95053E3B80CA1427D8263B4F4A254CB31FCF0BE280CA9B179C1C64BD2CF91D291C8FA73E9C8B9020C53162B59117FA54C5DD2B6EC26B5CB47DAFDC29767430F9DABA8BF12A265659B1871936CE9FA17279A2C8C61BC941D893B4858ED3A68834FA1261DF2B4A05011CBDB3E3DADAE9AF12F592C4240569031813DEC6063E8476E318D03DD1D29E895BF7ACC16C0ED8FAD74A2028A01D83B86D81F3DD5E094E4055DCEA040738F8778471A83E689F13D44D8ADE535DD5B543F154443E18C8A3FAE5F01D0A6B13A4B162676428388AFCEE06F7E4EFAEA9483F8E62F7E54CD436C8FE7DA4A1B508EE8FF076730C15E212A12D2627404BF19721CD71D531315ABA386DDD7928E321DB8DB94E3183968EAC104DA495BB5AA633DB2CFAB38D9F965B06FFDF41A96883663876095AFB7F7E9C9964ECE21380139B73250978FA680F62DEAE4389A0D874EFE0CE6001556C347BF8F617C39C42491220EDBC692147F111B3A521271E94E7605D733C3589744D24AC56117E5559419D173BF87331D4D2ACA4840A556DDFEFF8997B6151B3E8DF5D2BA3E4ED1219272893D5A410E6E8B3D8FD5397B4F86283A290F39B8F2CE41DB27CC8D26209429BB533AAFB6**

### Additional information

```js
localhost:3000/paymentredirect/@77539A0978176A5160ACDF87F51A9D6C2AC1A6885732571B87E2D20AE7718F19D0FDD32388B67377941D91E791934C6D1F064DE20F445E05BCC4B7BA6539C93A355C7641C3740192C1DD6038841BFB0C26B7FB01D1A4543616768D2191D6F7B86B853B6E90882EC27A1FCF1E9834E02955443BED9318947FDC4F3EF4CFCD546BE7DB3D782C87923DDBD2C0035AD52C6368D0A0E9B5CDF5673406E642264250A3ACA5F192EAE4B85494E6521DC0E873B8E9B2508A4AAA73A6C004EF7BB45A2DDB4ECA5BEC72F95053E3B80CA1427D8263B4F4A254CB31FCF0BE280CA9B179C1C64BD2CF91D291C8FA73E9C8B9020C53162B59117FA54C5DD2B6EC26B5CB47DAFDC29767430F9DABA8BF12A265659B1871936CE9FA17279A2C8C61BC941D893B4858ED3A68834FA1261DF2B4A05011CBDB3E3DADAE9AF12F592C4240569031813DEC6063E8476E318D03DD1D29E895BF7ACC16C0ED8FAD74A2028A01D83B86D81F3DD5E094E4055DCEA040738F8778471A83E689F13D44D8ADE535DD5B543F154443E18C8A3FAE5F01D0A6B13A4B162676428388AFCEE06F7E4EFAEA9483F8E62F7E54CD436C8FE7DA4A1B508EE8FF076730C15E212A12D2627404BF19721CD71D531315ABA386DDD7928E321DB8DB94E3183968EAC104DA495BB5AA633DB2CFAB38D9F965B06FFDF41A96883663876095AFB7F7E9C9964ECE21380139B73250978FA680F62DEAE4389A0D874EFE0CE6001556C347BF8F617C39C42491220EDBC692147F111B3A521271E94E7605D733C3589744D24AC56117E5559419D173BF87331D4D2ACA4840A556DDFEFF8997B6151B3E8DF5D2BA3E4ED1219272893D5A410E6E8B3D8FD5397B4F86283A290F39B8F2CE41DB27CC8D26209429BB533AAFB6
```

**Comment by devjiwonchoi** [comment]
What is your expected URL?

**Reply by junaidfactscomputer** [reply]
localhost:3000/paymentredirect/@77539A0978176A5160ACDF87F51A9D6C2AC1A6885732571B87E2D20AE7718F19D0FDD32388B67377941D91E791934C6D1F064DE20F445E05BCC4B7BA6539C93A355C7641C3740192C1DD6038841BFB0C26B7FB01D1A4543616768D2191D6F7B86B853B6E90882EC27A1FCF1E9834E02955443BED9318947FDC4F3EF4CFCD546BE7DB3D782C87923DDBD2C0035AD52C6368D0A0E9B5CDF5673406E642264250A3ACA5F192EAE4B85494E6521DC0E873B8E9B2508A4AAA73A6C004EF7BB45A2DDB4ECA5BEC72F95053E3B80CA1427D8263B4F4A254CB31FCF0BE280CA9B179C1C64BD2CF91D291C8FA73E9C8B9020C53162B59117FA54C5DD2B6EC26B5CB47DAFDC29767430F9DABA8BF12A265659B1871936CE9FA17279A2C8C61BC941D893B4858ED3A68834FA1261DF2B4A05011CBDB3E3DADAE9AF12F592C4240569031813DEC6063E8476E318D03DD1D29E895BF7ACC16C0ED8FAD74A2028A01D83B86D81F3DD5E094E4055DCEA040738F8778471A83E689F13D44D8ADE535DD5B543F154443E18C8A3FAE5F01D0A6B13A4B162676428388AFCEE06F7E4EFAEA9483F8E62F7E54CD436C8FE7DA4A1B508EE8FF076730C15E212A12D2627404BF19721CD71D531315ABA386DDD7928E321DB8DB94E3183968EAC104DA495BB5AA633DB2CFAB38D9F965B06FFDF41A96883663876095AFB7F7E9C9964ECE21380139B73250978FA680F62DEAE4389A0D874EFE0CE6001556C347BF8F617C39C42491220EDBC692147F111B3A521271E94E7605D733C3589744D24AC56117E5559419D173BF87331D4D2ACA4840A556DDFEFF8997B6151B3E8DF5D2BA3E4ED1219272893D5A410E6E8B3D8FD5397B4F86283A290F39B8F2CE41DB27CC8D26209429BB533AAFB6

**Comment by devjiwonchoi** [comment]
I've missed out that you've hosted it.

> when redirecting during testing it was working but after hosted giving error Bad Request - Invalid URL

Modify your [`hostname`](https://en.wikipedia.org/wiki/Hostname) to the hosted server IP or domain name.

For example, `example.com/paymentredirec/@...`

Also, it's better to add the HTTP protocol such as `http://` or `https://`.
